### PR TITLE
#162170682 Get Notifications

### PR DESCRIPTION
**Pull request by marcdomain**
#### What does this PR do?
Gives users the ability to view notifications

#### Description of Task to be completed?
`GET /notifications`
- Test the feature using `jest` and `react-testing-library`
- Create component for displaying notifications
- Consume the API endpoint
- Implement pusher-js to get real-time notifications

#### How should this be manually tested?
- Clone this repository `git@github.com:andela/balder-ah-frontend.git`
- cd into the directory on your local machine
- On your command line, run `npm install` to install the project npm packages
- Run `npm run start:dev` to start the app
- On your web browser, login to the app and visit  `/notifications` to view your notifications
- Watch out for pop ups of real-time notifications
- Run `npm test` to test the app
- Run `npm test -- --coverage` to view the test coverage

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
[#162170682](https://www.pivotaltracker.com/story/show/162170682)

#### Screenshots
screen shot 2019-02-01 at 16 03 46 

screen shot 2019-02-01 at 16 04 15

screen shot 2019-02-01 at 16 01 10

### CODE REVIEW

File: public/index.html [hunk 1]
```diff
@@ -50,5 +50,6 @@
       To begin the development, run `npm start` or `yarn start`.
       To create a production bundle, use `npm run build` or `yarn build`.
     -->
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/pusher/4.3.1/pusher.min.js"></script>
```
Comment (akhilome) [inline public/index.html]: This is no longer needed since we are using the npm package

File: src/App.jsx [hunk 2]
```diff
@@ -18,11 +21,35 @@ import NotFoundPage from './components/404/NotFoundPage';
...
 class App extends Component {
   async componentDidMount() {
-    const { getLoggedInUser: getUser } = this.props;
+    const { getLoggedInUser: getUser, newNotification: dispatchNewNotfication } = this.props;
     await getUser();
+
+    const notifications = JSON.parse(localStorage.getItem('newNotification'));
+    dispatchNewNotfication(notifications || []);
+
+    const pusher = new Pusher('00e0c702ebdfbdfe5985', {
```
Comment (akhilome) [inline src/App.jsx]: This is a secret key hence should be in the `.env` file.

File: src/App.jsx [hunk 3]
```diff
@@ -43,6 +70,7 @@ class App extends Component {
             <Route path="/search" component={SearchResult} />
             <Route path="/reset-password" component={PasswordReset} />
             <Route path="/update-password" component={UpdatePassword} />
+            <Route path="/notifications" component={GetNotifications} />
```
Comment (akhilome) [inline src/App.jsx]: This should be a `<PrivateRoute />`

File: src/App.test.jsx [hunk 4]
```diff
@@ -6,6 +6,16 @@ import axios from './utils/axiosInstance';
 
 const axiosMock = new MockAdapter(axios, { delayResponse: 500 });
 
+// describe('Notifications', () => {
```
Comment (akhilome) [inline src/App.test.jsx]: ???

File: src/components/notifications/Notifications.jsx [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+};
+
+GetNotifications.propTypes = {
+  getNotifications: PropTypes.func.isRequired,
+  notifications: PropTypes.oneOfType([PropTypes.array]),
+};
+
+const mapStateToProps = ({ auth: { isLoggedIn }, notifications: { notifications } }) => ({
+  notifications,
+  isLoggedIn,
```
Comment (akhilome) [inline src/components/notifications/Notifications.jsx:61]: This was not used anywhere in this file.

**Review by akhilome — COMMENTED** [review]
Great job so far Marcus.

I'm aware this is still `WIP` but here are some quirks I spotted while going over the PR just incase you miss them before badging the PR `ready`

File: src/App.jsx (hunk 2 above)
Comment (oluwajuwon) [reply src/App.jsx]: @marcdomain I agree with @akhilome

File: src/App.jsx [hunk 6]
```diff
@@ -18,11 +21,35 @@ import NotFoundPage from './components/404/NotFoundPage';
...
+
+    const notifications = JSON.parse(localStorage.getItem('newNotification'));
+    dispatchNewNotfication(notifications || []);
+
+    const pusher = new Pusher('00e0c702ebdfbdfe5985', {
+      cluster: 'us2',
+      encrypted: true,
+    });
+
+    const channel = pusher.subscribe('balderah-notifications');
```
Comment (oluwajuwon) [inline src/App.jsx:40]: hi @marcdomain please can you explain to me what is happening here? to line 41

**Review by oluwajuwon — COMMENTED** [review]
nice work @marcdomain I left some comments on your PR, please go through them

**Comment by Johnsonojo** [comment]
@marcdomain, nice job done on the implementation of this feature. It looks good to me. Just make requested changes.

**Review by frostyblok — APPROVED** [review]

File: src/App.jsx (hunk 6 above)
Comment (marcdomain) [reply src/App.jsx:40]: The string `balderah-notifications` is a pusher channel that Kizito had already implemented on the backend. So to connect to it from an client app, the pusher must be subscribed to the same channel (using the string above). Prior to that an instance of pusher must be created using the same `pusher key` used on the backend.

**Pull request merged**